ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -302,14 +307,64 @@ func convertToECSVolumes(hostPaths *adapter.Volumes) []*ecs.Volume {
 			}}
 		output = append(output, ecsVolume)
 	}
-	// volumes with an empty host path
-	for _, volName := range hostPaths.VolumeEmptyHost {
+
+	// process docker volumes
+	dockerVolumes := make(map[string]DockerVolume)
+	if ecsParams != nil {
+		for _, dockerVol := range ecsParams.TaskDefinition.DockerVolumes {
+			if dockerVol.Name != "" {
+				dockerVolumes[dockerVol.Name] = dockerVol
+			}
+		}
+	}
+
+	// volumes without host path (allowed to have Docker Volume Configuration)
+	volumesWithoutHost, err := mergeVolumesWithoutHost(hostPaths.VolumeEmptyHost, ecsParams)
+	if err != nil {
+		return nil, err
+	}
+	output = append(output, volumesWithoutHost...)
+	return output, nil
+}
+
+func mergeVolumesWithoutHost(composeVolumes []string, ecsParams *ECSParams) ([]*ecs.Volume, error) {
+	volumesWithoutHost := make(map[string]DockerVolume)
+	output := []*ecs.Volume{}
+
+	for _, volName := range composeVolumes {
+		volumesWithoutHost[volName] = DockerVolume{}
+	}
+
+	if ecsParams != nil {
+		for _, dockerVol := range ecsParams.TaskDefinition.DockerVolumes {
+			if dockerVol.Name != "" {
+				volumesWithoutHost[dockerVol.Name] = dockerVol
+			} else {
+				return nil, fmt.Errorf("docker_volume.name is a required field")
```

Comment:
Since `docker_volume.name` is not the name of a specific [ECS volume field](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/docker-volumes.html) (and the closest `volume.name` is listed as not required), can we rephrase this err msg to something like: "Name is required when specifying a docker volume" ?

Reply:
Sure, I'll change it to that. I was thinking that user's would understand that `docker_volume.name` is the ECS Params yaml.